parsing/dialects/intermediate/dialects/dialect1/parserImpls.ftl

```diff
@@ -5049,3 +5051,48 @@ SqlCloseCursor SqlCloseCursor() :
     <CLOSE> cursorName = SimpleIdentifier()
     { return new SqlCloseCursor(s.end(this), cursorName); }
 }
+
+SqlSelectAndConsume SqlSelectAndConsume() :
+{
+    boolean set = false;
+    final List<SqlNode> selectList;
+    final SqlNodeList parameters = new SqlNodeList(getPos());
+    final SqlIdentifier fromTable;
+    final int topNum;
+    final Span s = Span.of();
+    SqlNode e;
+}
+{
+    <SELECT>
+    [ <SET> { set = true; } ]
+    <AND> <CONSUME> <TOP> topNum = IntLiteral() {
+        if (topNum != 1) {
+            throw SqlUtil.newContextException(getPos(),
+                RESOURCE.numberLiteralOutOfRange(String.valueOf(topNum)));
+        }
+    }
+    selectList = SelectList()
+    <INTO>
+    (
+        e = SimpleIdentifier()
+    |
+        e = SqlHostVariable()
+    )
+    { parameters.add(e); }
+    (
+        <COMMA>
+        (
+            e = SimpleIdentifier()
+        |
+            e = SqlHostVariable()
+        )
+        { parameters.add(e); }
+    )*
+    <FROM>
+    fromTable = SimpleIdentifier()
```

Comment:
Can this be a CompoundIdentifier?

Reply:
The docs don't explicitly say it but I think it can be so I'll update it.